src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -270,7 +318,96 @@ export const NewProposal = ({ navigation, data, query }) => {
   );
 };
 
+const ImageSelector = ({ control, field, item, selectImage }) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [imageInfoText, setImageInfoText] = useState('');
+
+  const { buttonTitle, infoText } = item;
+  const { name, onChange, value } = field;
+
+  return (
+    <>
+      <Input
+        {...item}
+        control={control}
+        errorMessage={errorMessage}
+        hidden
+        validate
+        name={name}
+        value={value}
+      />
+      <RegularText smallest placeholder>
+        {infoText}
+      </RegularText>
+
+      {value ? (
+        <>
+          <WrapperRow center spaceBetween>
+            <Image source={{ uri: value }} style={styles.image} />
+
+            <TouchableOpacity
+              onPress={() => {
+                onChange('');
+                setErrorMessage('');
+                setImageInfoText('');
+              }}
+            >
+              <Icon.Trash color={colors.error} size={normalize(16)} />
+            </TouchableOpacity>
+          </WrapperRow>
+          {!!imageInfoText && <RegularText smallest>{imageInfoText}</RegularText>}
+        </>
+      ) : (
+        <Button
+          title={buttonTitle}
+          invert
+          onPress={async () => {
+            const { uri, type } = await selectImage();
+            const { size } = await FileSystem.getInfoAsync(uri);
+            const errorMessage = imageErrorMessageGenerator({
+              size,
+              imageType: uri.substr(uri.length - 4)
+            });
+
+            setErrorMessage(texts.consul.startNew[errorMessage]);
+            setImageInfoText(`(${type}/${uri.substr(uri.length - 3)}, ${bytesToSize(size)})`);
+            onChange(uri);
+          }}
+        />
+      )}
+    </>
+  );
+};
+
+const bytesToSize = (bytes) => {
+  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
+  if (bytes == 0) return '0 Byte';
+  let i = parseInt(Math.floor(Math.log(bytes) / Math.log(1024)));
+
+  return Math.round(bytes / Math.pow(1024, i), 2) + ' ' + sizes[i];
+};
+
+const imageErrorMessageGenerator = ({ size, imageType }) => {
+  const isJPG = imageType === '.jpg' ?? 'jpeg';
```

Comment:
`.` is still missing in `jpeg`, right?

Reply:
`.` is not missing. I have the last 4 letters of the uri data.

jpg = `.jpg`
jpeg = `jpeg`

Reply:
i understand. it is related to the `- 4`. this is too specific. let's think of some RegEx that will help us in all situations.

Reply:
96320ff2f0d88d127a9a420ef721f9dc1b50613c